packages/workflow-core/src/lib/workflow-runner.ts

```diff
@@ -744,11 +744,22 @@ export class WorkflowRunner {
         responseBody,
         apiPlugin.persistResponseDestination,
       );
-    } else {
-      this.context.pluginsOutput = {
-        ...(this.context.pluginsOutput || {}),
-        ...{ [apiPlugin.name]: responseBody ? responseBody : { error: error } },
-      };
+    }
+
+    if (!apiPlugin.persistResponseDestination && responseBody) {
+      this.context = this.mergeToContext(
+        this.context,
+        responseBody,
+        `pluginsOutput.${apiPlugin.name}`,
+      );
+    }
+
+    if (error) {
+      this.context = this.mergeToContext(
+        this.context,
+        { name: apiPlugin.name, error, status: ProcessStatus.ERROR },
```

Comment:
Main changes are `name` and `status`.

Reply:
@Omri-Levy didn't we said we are going to remove the dependency on `plugin.name` and use the key instead?

Reply:
or its part of the second phase